arceos-vmm/build.rs

```diff
@@ -75,11 +77,118 @@ fn open_output_file() -> fs::File {
         .write(true)
         .create(true)
         .truncate(true)
-        .open(&output_file)
+        .open(output_file)
         .unwrap()
 }
 
+fn read_toml_file(file_path: &str) -> io::Result<Value> {
+    println!("Reading {}", file_path);
+    let contents = fs::read_to_string(file_path)
+        .unwrap_or_else(|_| panic!("Failed to read file {}", file_path));
+    let parsed_toml: Value = contents
+        .parse::<Value>()
+        .expect("failed to parse config file");
+    Ok(parsed_toml)
+}
+
+/// generate function to load guest images from config
+/// Toml file must be provided to load from memory.
+fn generate_load_guest_img(
+    mut out_file: fs::File,
+    config_toml_path: Option<Vec<OsString>>,
+) -> io::Result<()> {
+    // Convert relative path to absolute path
+    fn convert_to_absolute(configs_path: &str, path: &str) -> PathBuf {
+        let path = Path::new(path);
+        let configs_path = Path::new(configs_path).join(path);
+        if path.is_relative() {
+            fs::canonicalize(&configs_path).unwrap_or_else(|_| path.to_path_buf())
+        } else {
+            path.to_path_buf()
+        }
+    }
+
+    if let Some(config_path) = config_toml_path {
+        // Started from the first config item by default.
+        if let Some(guest_config) = config_path.first() {
```

Comment:
do you consider if multiple VM config files are privided, and we have to generate multiple `load_xxx` functions for each VM.

It seems that you only load the first one.

Reply:
we can use a array here, with `vm_id` as index?

something like `get_kernel_binaries_of_vm(index: usize) -> Option<&'static [u8]>`